### DISCUSSION TITLE
New Website, Logo, Playground etc

### DISCUSSION THREAD

**Discussion by phpnode** [peggyjs/peggy#55] [Ideas]
The website is based on the original pegjs.org site and it's pretty dated, it would be nice to have a new site with some better branding. It's also an opportunity to introduce a nicer playground experience which is really important for people new to the project (and also for people not so new to it). TypeScript, ASTExplorer, Babel all have pretty nice playgrounds that we can draw inspiration from.

**Comment by hildjj** [comment]
I converted this to a discussion, because I think it will end up being project-sized.

**Comment by Mingun** [comment]
Also we should consider to switch to some static site generator instead of manually edit `docs/documentation.html`

**Reply by hildjj** [reply]
See #152

**Comment by hildjj** [comment]
See #156

**Comment by reverofevil** [comment]
I've made a monaco-based alternative to `/online` page, and probably could take part in this.

**Comment by hailiang-wang** [comment]
Really glad to see pegjs is moving forward, and keep on coming with peggy!
Switch to peggy now.
You all make great works. Thanks.

**Comment by jcubic** [comment]
Do you want a logo? I can see if I can make something. I'm a developer but I sometimes design logos and logotypes. I don't do that professionally because I do not always come up with a good design.

Here are a few of my logos: https://jakub.jankiewicz.org/logo.php

**Reply by hildjj** [reply]
> Do you want a logo? I can see if I can make something. I'm a developer but I sometimes design logos and logotypes. I don't do that professionally because I do not always come up with a good design. 

That sounds like fun!  Sure, let's see what you come up with.

**Reply by reverofevil** [reply]
Oh wow, those logos look really good!

**Comment by jcubic** [comment]
I came up with a few designs, I was thinking about what image the parser generator should have and I only came up with Cog Wheel so all logos are based on that. The last one was my attempt to create P out of Cog wheel but it doesn't look very good.

1:
![A](https://user-images.githubusercontent.com/280241/174486536-2608f96e-ce21-45c0-96f4-b48db2de4691.png)
2:
![B](https://user-images.githubusercontent.com/280241/174486539-c6fa3101-39e1-42cb-852c-56e5296d6f7d.png)
3:
![C](https://user-images.githubusercontent.com/280241/174486597-401cfe5c-d28d-4728-bc83-a85b920b481c.png)
4:
![D](https://user-images.githubusercontent.com/280241/174486542-8cda71cb-6348-4e59-a7c7-254d39c24e5b.png)
5:
![E](https://user-images.githubusercontent.com/280241/174486543-d35b074c-d440-478a-9b9e-6a0df6f403ef.png)

**Reply by Mingun** [reply]
May be try to stylize G[ear]G[ear] as two gears?

**Reply by jcubic** [reply]
Will try I have two gears version in my notebook, but they were not for GG.

**Reply by reverofevil** [reply]
Logos usually convey the message, and these logos unfortunately convey the message Peggy is unusable. :D

Could you please consider using [proper gear profile](https://en.wikipedia.org/wiki/Involute_gear)?

**Reply by jcubic** [reply]
@polkovnikov-ph sure. Will update the logos. Do you like any particular one if it had a proper gear profile?

**Reply by reverofevil** [reply]
@jcubic I think the second is best, because it allows to remove text from it in case someone decides to use it as favicon, repository logo etc.

**Reply by jcubic** [reply]
@polkovnikov-ph It's also my favorite. I can make that proper gear and also create GG as gear.

**Reply by hildjj** [reply]
I like the second just fine.  I wonder if there's anything you can do with `{}` in there...

**Reply by jcubic** [reply]
Do you mean like `{ P }` characters?

**Reply by hildjj** [reply]
Maybe.  What if the `{}` were elongated horizontally into something close to a circle, and used as the inside surface of the gear.  They could just be negative space, for instance.

**Reply by jcubic** [reply]
![F](https://user-images.githubusercontent.com/280241/174497785-9dc66285-4610-4968-bf2c-a873ca74e20a.png)
![G](https://user-images.githubusercontent.com/280241/174497786-e5d7d6f9-5358-4cb6-b4df-67dc60e1050e.png)

For me, it doesn't look good. Curly brackets don't match the gear. But I can see if I can use just the curly brackets, it was my original idea but I actually didn't use it.

**Reply by hildjj** [reply]
I defer to your expertise.
